### ISSUE TITLE
One-week sprint plan

### ISSUE THREAD

**Issue by bhaugen** [FreedomCoop/valuenetwork#131]
Should we plan a one-week sprint? I think so. It means figuring out what work should be included, and who will do it.

### Candidates for issues to include:
* https://github.com/FreedomCoop/valuenetwork/issues/128 Make membership request comments more visible to Coop Admins
* https://github.com/FreedomCoop/valuenetwork/issues/123 Should we add project admin functions to the work app or add more permissions to the admin app? (Or create a new Projects app.) 
    * This job would include making the decision and doing as much of the work as we think can be completed in a week.
    * It might also include whatever is left to do on https://github.com/FreedomCoop/valuenetwork/issues/108 New page "Your Projects" 
    * And also maybe https://github.com/FreedomCoop/valuenetwork/issues/107 Create an application process for using services or join groups
* https://github.com/FreedomCoop/valuenetwork/issues/121 Decline and delete for membership requests
* https://github.com/FreedomCoop/valuenetwork/issues/127 Add a forum app to OCP

Please suggest which issues should be added, removed, or changed.

### Decisions to be made:
* Whether to do a planned sprint, or keep working on issues one-by-one.
* What issues to include in the sprint.
* Whether to include project admin functions inn the work app (as started in the new page "Your Projects") or create a new Project app. I am guessing from the chatter that people prefer not to try to do the project management in the coop admin app, but that would still be a possibility.
    * See also https://github.com/FreedomCoop/valuenetwork/issues/130 should resource types be tied to contexts.
* What scope for this set of features for a one-week sprint. The features seem to be revolve around project manager requirements.
* Who will do what.

**Comment by XaviP** [comment]
Maybe this issue can be included too: ["Other" fields](https://github.com/FreedomCoop/valuenetwork/issues/64)
I mostly prefer to work on forum or comments improvements.

**Comment by bhaugen** [comment]
> I mostly prefer to work on forum or comments' improvements.

You got 'em!

I'll pick from what nobody else wants.

**Comment by bhaugen** [comment]
Tentative plan for tomorrow?
 How about if Lynn and I start a new branch called project, and create a new app called project. There's a lot of setup involved in a new app, and Lynn has done it recently.

Then @XaviP can take off on the comments improvements and forum maybe in another branch or two. The comments improvements most likely go into the coop admin app, and the forum will most likely be its own new app.

Then when Lynn and I get the new project app started, @bum2 can start on the menus and bring his "your projects" page over. In the meantime, Bumbum, maybe you can look at that page and try to figure out why it is landing people back in the coop admin app. And if you figure that out, look at what the valueaccounting/agent page is doing to prevent duplicate agent IDs.

As always, these are just some ideas. You may have better ones.

**Comment by bum2** [comment]
I still don't see the need of forums, and the comments feature we already have it and is working. The improvements about comments i think are: 
- notification to project managers (when a user make a comment).
- notification to user (when managers or admins make a comment).
- simple tasks for project managers about 'pending to answer' (or view?) user comments (with a link to it). Also can be a special page or a block of 'pending comments' to insert in a dashboard, but i think is better just 'tasks' for project managers.
I think this improvements should go to the work app interface (not the coop admin app) for 'project managers' (i prefer to reword to 'coordinators'). So we develop that really needed interface. This means setting the 'manager-coordinator' relation (association) correctly to the responsable users of every group (minimun one coordinator for every context-agent or project), so they see all this management tools in the work app, in the 'your projects' page (only if they are coordinators-managers of a group).

In the page 'your projects' is where i need help for:
- adding the 'create new project' part:
    - if member is collective (the project): the created agents (projects) are like child projects (sub-agents) of the user-context-agent.
    - if member is individual: the created projects became 'parent context-agents' of the user.
    - in both cases, the member-user creating the project becames the default project coordinator (manager) of the new project.
- adding project participants:
    - if project access is 'moderated': 
        - if we can have the 'flexible forms' plugin: an admin is needed to setup the special form, and set the form id for the project.
        - if we use a generic form: a system for defining the 'questions' strings (field names) of 3 or more customizable text fields of the generic form.
        - in both cases, many of the fields are common required fields for ocp or for any project. Also a introduction text for every project form is needed. And also is needed a way of translating such strings (coming from the db, like the help strings).
        - if project visibility is 'public': the request access form is available from non-logged visitors (needs captcha).
        - if project visibility is 'FCmembers': the request access form is available only for FC members, and we need to show a list of joinable groups somewhere in members UI (new page 'Join a project'), this list opens each 'join form' page in the work app.
        - if project visibility is 'private': the group doesn't show nowhere (except for project members and admins).
    - if the project access is 'autojoin': no need form, only show the project in a 'Join a project' page for any ocp user or only for FC members (they can autojoin).
- design and code the sub-pages of 'your projects' page for coordinators (managers): contributions, active orders, exchanges, etc... see #123 and #108 
- create the coordinators workflow for plan process tasks in a project: huge subject to design and work-on from the member's point of view, and include all needed pages with restricted scope in the work app.

In some comments i'm listing the needed tasks, but if is not clear enough, then i will take the time to create a special simple issue for every one of the steps and tasks i'm explaining. For me is easier to write the plans all together (like i'm doing), so we see the general picture, but perhaps is not easy for others to just pick a single text line of the lists and treat it as a pending issue or a task itself (they are). 
I though it was better to avoid many small issues in github, and try to group them in big issues or subjects, each one with a list plan. Perhaps we can do both: big issues with a plan listing and also small issues for every step or pending task.

**Comment by bum2** [comment]
If the extension of the EconomicAgent model (to include project options like visibility or access style and related data) is not available this week, we can do all the other things using the most urgent use cases: the CVN project needs to be 'moderated' and 'public' (external request form).
I'm not sure what is needed for 'dezentrale'... perhaps is also moderated but not public access?

**Comment by bhaugen** [comment]
@bum2 do you agree with creating a new project app? Or not? (Wasn't clear from trying to read through the lines of a couple of your comments.)

**Comment by bum2** [comment]
My answer was "yes, let's do it" and trying to figure out its name, in the comment (3 days ago) at #107...
Anyway the user interface and the main url '/work/' is the same one (no need a special url slug or interface for projects)... the user's interface we are calling 'work app' but in reality it is the 'work GUI'.
I take this opportunity to say i don't like to use of the 'work' string, but in this case (urls) i don't have a better proposal yet.
All the user interfaces (participants, candidates, members and project managers) are in the same pages and templates (inside /work/ folder). The menus and pages are showing diferent things depending on the user role.
If you prefer to work all the 'projects' stuff and functions in a new django App, for me is ok, i will get into. But the interface part (menus and pages) is already inserted in the 'work' django App (with filters for members or not, for managers or not, etc), and the easy for me is to keep developing the GUI there.

**Comment by bhaugen** [comment]
> My answer was "yes, let's do it" and trying to figure out its name, in the comment (3 days ago) at #107...

Ok, I'll follow up on that in my next comment in this thread.

> Anyway the user interface and the main url '/work/' is the same one (no need a special url slug or interface for projects)... the user's interface we are calling 'work app' but in reality it is the 'work GUI'.
I take this opportunity to say i don't like to use of the 'work' string, but in this case (urls) i don't have a better proposal yet.

The string in the URLs is called out in the top-level urls.py, for example:
```
    url(r"^accounting/", include("valuenetwork.valueaccounting.urls")),
    url(r"^admin/", include(admin.site.urls)),
    url(r"^account/", include("account.urls")),
    url(r"^notification/", include("notification.urls")),    
    url(r"^work/", include("work.urls")),
    url(r"^api/", include("valuenetwork.api.urls")),
```

Might be able to change it right there, but I haven't tried, and don't understand all the repercussions if we do.

**Comment by bhaugen** [comment]
Assuming we will start a new app called project, I'll start working on it now, with some help from Lynn.

One of the possibilities, given that it will have a new base template, is we can either move to bootstrap 3 (we're still on bootstrap 2), or some other CSS framework, or ditch CSS frameworks altogether and roll our own. Anybody got any opinions or recommendations?

Anything else that should go in the base template (to be inherited by every other template in the new project app)?

**Comment by XaviP** [comment]
> I still don't see the need of forums
Ok, postponed.

> notification to project managers (when a user make a comment).
> notification to user (when managers or admins make a comment).
These two points is what I'm working on. The creation of tasks for admins is out of this sprint plan. Postponed.

If we increase the email notifications, we will need a subscribe/desubscribe system too.

**Comment by fosterlynn** [comment]
from @bhaugen :

>Assuming we will start a new app called project, I'll start working on it now, with some help from Lynn.

from @bum2 :

>If you prefer to work all the 'projects' stuff and functions in a new django App, for me is ok, i will get into. But the interface part (menus and pages) is already inserted in the 'work' django App (with filters for members or not, for managers or not, etc), and the easy for me is to keep developing the GUI there.

I think based on @bum2, we are actually not going to make a new project Django app.  It doesn't make any sense to split the UI part and the backend.  We will just continue in the work app.

**Comment by bum2** [comment]
we don't need a new base tamplate, the one we are working with is enough (work/templates/new_base.html), so, if needed a new one, is just a copy of it (empty extension).

**Comment by bhaugen** [comment]
> If we increase the email notifications, we will need a subscribe/desubscribe system too.

We already got one. It's in your settings (on the top-right user dropdown menu). Look for notification settings. They are now all opt-out, but I think we can change that to opt-in.

We can also set up an admin association type and make the people who should get these notifications admins of whatever group is responsible for whatever the notification is about. That would cut down the email distribution list a lot and maybe opt-out would be more acceptable?

**Comment by bhaugen** [comment]
Looks to me like we just dropped the new project app out of the one-week sprint plan.

So what's left, and who is working on what?

**Comment by bhaugen** [comment]
@XaviP is working on comments, right? If you want to use email notifications, let me know and I'll do the setup for them. If you are starting a new branch for that, I'll work in your branch.

@bum2 I assume you are working on "your projects" and its various extensions? You listed [here](https://github.com/FreedomCoop/valuenetwork/issues/107#issuecomment-242877646) some things you wanted help with. Is that list up-to-date? If not, please let me know.

I figured I would plug in on whatever was left over. I think I would start with whatever anybody else wants help with, and then move on to other issues.

**Comment by bhaugen** [comment]
@fosterlynn proposes that the first thing we do is add a new Project model to work/models.py with a one-to-one foreignkey to EconomicAgent, with the new fields proposed by @bum2 

We would do that in a new project branch (unless @bum2 you have already created a new branch, in which case we will work in your branch).

**Comment by bum2** [comment]
I thought it was good to do the new model Project in a new app... but if is easier, let's do it in the work app. Can you take that part Bob? is the extension of the EconomicAgent model with the stuff only for projects: basically the visibility and access settings.
After that, you can choose a good 'flexible forms' plugin and try to make it work as a separate app (will be for admins only, i don't find a user customizable forms plugin) and add the form_id field in the Project model (to link each project to it's custom form for moderated access). 
If we don't use the forms plugin, we will need more fields to offer a generic form customization (some 'asked questions') and a form intro text.
All this is only for the projects that choose 'moderated' access...

**Comment by fosterlynn** [comment]
>In the page 'your projects' is where i need help for:
- adding the 'create new project' part:
    - if member is collective (the project): the created agents (projects) are like child projects (sub-agents) of the user-context-agent.
    - if member is individual: the created projects became 'parent context-agents' of the user.
    - in both cases, the member-user creating the project becames the default project coordinator (manager) of the new project.
- adding project participants:
    - if project access is 'moderated': 
        - if we can have the 'flexible forms' plugin: an admin is needed to setup the special form, and set the form id for the project.
        - if we use a generic form: a system for defining the 'questions' strings (field names) of 3 or more customizable text fields of the generic form.
        - in both cases, many of the fields are common required fields for ocp or for any project. Also a introduction text for every project form is needed. And also is needed a way of translating such strings (coming from the db, like the help strings).
        - if project visibility is 'public': the request access form is available from non-logged visitors (needs captcha).
        - if project visibility is 'FCmembers': the request access form is available only for FC members, and we need to show a list of joinable groups somewhere in members UI (new page 'Join a project'), this list opens each 'join form' page in the work app.
        - if project visibility is 'private': the group doesn't show nowhere (except for project members and admins).
    - if the project access is 'autojoin': no need form, only show the project in a 'Join a project' page for any ocp user or only for FC members (they can autojoin).
- design and code the sub-pages of 'your projects' page for coordinators (managers): contributions, active orders, exchanges, etc... see #123 and #108 
- create the coordinators workflow for plan process tasks in a project: huge subject to design and work-on from the member's point of view, and include all needed pages with restricted scope in the work app.

@bum2 I can do some work on some of this.  Can you assign me something please?  I can work max half time on FC this week, so assign me something with not too many dependencies if you can.

**Comment by bhaugen** [comment]
I created and pushed the new project branch. Will create and test new Project model.

**Comment by bhaugen** [comment]
> I thought it was good to do the new model Project in a new app..

The problem I see is you want the UI for the new app to be in the work app, which I think will be complicated and confusing.

**Comment by fosterlynn** [comment]
>@bum2 I can do some work on some of this. Can you assign me something please? I can work max half time on FC this week, so assign me something with not too many dependencies if you can. 

It would be best if I work on something that doesn't involve all the new membership code, which I am not up to speed on.  So maybe something at the end of your list?

**Comment by bum2** [comment]
You can create a new branch and i will work on it from now on, my last branch was merged and deleted... my last commits were on master directly (not sure if deployed to prod).

Lynn you can help me finding or defining a way for translating db strings (like help texts, custom project texts, or any other non-hardcoded texts). Also i will love to have help figuring out what should appear in the 'your projects' subpages for managers: contributions seems easy, and perhaps associations also, but 'active orders' or the 'create exchange' feature are still far from being defined in the members UI...

**Comment by bhaugen** [comment]
Let's make a hard-and-fast decision for this one-week sprint: should I start the project branch by creating a new project app, or put the new Project model in the work app?

If we want all the UI to go into the work app, it's better to keep working there and *not* create a new app.

**Comment by bum2** [comment]
Ok Bob, let's do a new model in the same work app!
Ok Lynn, you can help defining the last part of the list, related plan process, resources, orders, etc. (if you feel more confortable). Maybe you also like the search and test of a good 'flexible forms' plugin? or the 'how to translate db strings' system?

**Comment by bhaugen** [comment]
> Ok Bob, let's do a new model in the same work app!

Okay, will do (in the project branch).

**Comment by fosterlynn** [comment]
>Ok Lynn, you can help defining the last part of the list, related plan process, resources, orders, etc. (if you feel more confortable). Maybe you also like the search and test of a good 'flexible forms' plugin? or the 'how to translate db strings' system?

Of those, I like the first the best, where I have more background and can probably help more.  So I will start on that for now: help defining the last part of the list, related plan process, resources, orders, etc.

**Comment by bum2** [comment]
Marvelous brothers! what a powerful team! i love you all...

**Comment by bum2** [comment]
I forget some parts of this list above, for projects with moderated access:
- A join-requests Table in the members_agent.html page, only when user is manager of that group.
- The comments page for the project candidates.
Both points are more defined in the #107 issue...

**Comment by bhaugen** [comment]
I put a first draft of the new Project model [here](https://github.com/FreedomCoop/valuenetwork/issues/107#issuecomment-243172109).

Please keep followup comments in that thread so I can find them.

**Comment by bhaugen** [comment]
I think I'll try a bit of multi-tasking and work on a model for [Other fields](https://github.com/FreedomCoop/valuenetwork/issues/64) at the same time...

(...always a bad sign when an old fart tries multi-tasking...)

**Comment by XaviP** [comment]
> The comments page for the project candidates.

Hey, let's wait to have this page running. :-)
We can clone it from work/templates/work/membership_request_with_comments.html
The notification can be a little bit harder, but thinking about it.

**Comment by bum2** [comment]
Yes XaviP, i was thinking to copy all that comments code to a new 'Join project request' page in the work app. This new comments page is only seen if the user is candidate of a project with moderated access. And if the user is candidate of more than one project with moderated access, then the menu can show a link to each instance of this new page, initially like: "{project_name} Access Request" items, and once accepted ('active' relation) the menu shows "{project_name} Feedback" items, but is the same page with the filled form request table* and the comments thread. (*by the way, let's remember to add one day some editing capabilities to this related request row table, so people can update their own data).
The whole notification part (and settings for project coordinators) of any instance of the comments tool is what is more undefined... and only for projects with moderated access.

**Comment by bhaugen** [comment]
Pushed new Project and SkillSuggestion models to github in project branch:
https://github.com/FreedomCoop/valuenetwork/blob/project/work/models.py#L73

**Comment by XaviP** [comment]
> The whole notification part (and settings for project coordinators) of any instance of the comments tool is what is more undefined... and only for projects with moderated access

This is my plan: the notification system works by a signal called "comment_was_posted" that fires when a comment is saved in db. We can control which content_type the comment is attached to. So, in case the content_type is membership_request, it will notify to node app admins (with subscribe system already working), and in case content_type is a (new) join_request for a project, it will notify to the admin of that project (I don't know yet how to use the subscribe system for this last notification).
.

**Comment by bum2** [comment]
Great Xavi! the notification in the last case is to the 'coordinators' of the project (managers) normally.
We can use the 'coordinator of' (= manager of) relation like if they are the 'admins' of a project. So all settings, management tools and notifications are for them. The main filter is: "if user in agent.managers:"

**Comment by bhaugen** [comment]
What should I do next, sprinters? If no suggestions, I'll pick something from the list and start working on it.

**Comment by bum2** [comment]
@bhaugen the Fobi forms are needed (if we're able to use them) or an alternative system (a generic join-request form) to add the 'form_id' FK field in the Project model (it can be Null). If the project chooses 'moderated' join_style, then we should add a sub-page in the members_agent.html (the 'project editor') to configure the Fobi form fields. Keep in mind that most of the fields for the join form are not custom, they are common for all new users (some of the OCP User fields and also some of the EconomicAgent fields), so the Fobi thing is only to define some project's custom fields.

**Comment by bhaugen** [comment]
As I said in Telegram, I'm working on the "Other" field now. I'll probably just do the bare bones to get it started, and then ask for opinions on what it needs next. After that, I figured I would connect with @XaviP on comment notifications.

Want to try to get Fobi into this week's sprint? Where do you think that fits into the priority queue?

**Comment by bhaugen** [comment]
While I was waiting for feedback, I started a fobi branch, and pushed it to github. Did more testing of fobi. Ran into one not very serious bug, but overall, looks pretty good. That's as far as I will go with it.

Will probably start working on https://github.com/FreedomCoop/valuenetwork/issues/121 Decline and delete for membership requests next, unless anybody has better ideas. 

Also, we should think about what we can deploy to testocp so Maro can do some testing...

**Comment by bhaugen** [comment]
We're also piling up several branches that are not merged with each other. @fosterlynn what would you advise us to do about that? Expect huge merge conflicts? Keep pulling master regularly and merging master with our branches?

**Comment by fosterlynn** [comment]
>We're also piling up several branches that are not merged with each other. @fosterlynn what would you advise us to do about that?

I wouldn't worry about it.  Hopefully we all are not working on the same parts of the code too much.  Also, it won't help to merge master to a branch unless master has the updates from other branches.  If some people think they are working in the same areas in the code, it might be good to merge those branches back and forth as we work, but coordinate that individually so code is reasonably stable, etc. 

Other thoughts welcome....

**Comment by bhaugen** [comment]
How are we doing, gang? It's the end of Tuesday for us. I'll be back in the morning, figuring out what to do next. If I can help either @bum2 or @XaviP in their work, that would be my priority. I'd rather help you all take over the programming work and just be your helper.

Otherwise, I will start working on on https://github.com/FreedomCoop/valuenetwork/issues/121 Decline and delete for membership requests next, unless anybody has better ideas.

**Comment by XaviP** [comment]
Comment notifications are working ok in testocp. Plaese, test it.
I've closed the issue about it. If any improvement needed, we can open a more concrete issue.

**Comment by bhaugen** [comment]
> Comment notifications are working ok in testocp. 

Are they merged into master now? And is testocp running on master?

I think I'd like to do a couple more comment visibility improvements, which I'll put in another issue:
https://github.com/FreedomCoop/valuenetwork/issues/136

**Comment by XaviP** [comment]
> Are they merged into master now? And is testocp running on master?
Yes, they are merged into master and deployed in testocp (master branch).
After some testing in testocp, I've created a new release (https://github.com/FreedomCoop/valuenetwork/releases/tag/v0.9.2) and deployed it in prod ocp.

**Comment by bhaugen** [comment]
It's Friday. Where are we at in our one-week sprint?

* #128 Make membership request comments more visible to Coop Admins [notifications done, I'll work on https://github.com/FreedomCoop/valuenetwork/issues/136 today]
* #123 Should we add project admin functions to the work app or add more permissions to the admin app? [ We're adding them to the work app. @bum2 - what's the state of this one?]
* #121 Decline and delete for membership requests [Done but still in the request_actions branch, which has a migration that needs to be merged with the project migrations]
* #64 "Other" fields [done, and I think merged into master]
* #127 Add a forum app to OCP [I think we decided to cut this from the sprint?]

**Comment by fosterlynn** [comment]
>#123 Should we add project admin functions to the work app or add more permissions to the admin app? [ We're adding them to the work app. @bum2 - what's the state of this one?]

I'm working on putting process logging in, will be done by Sunday.  (That's not a project function, but is predecessor to planning.)

**Comment by XaviP** [comment]
Hi, I think we have to make a new release and deploy to prod ocp with these features:
- Create/edit project button. (tested)
- Suggest skills in profile. (tested)
- Bug about unicode.
I'm just waiting for @bum2 confirmation to go ahead.
Please, don't merge into master nothing more until we've tested it enough, released and deployed.

**Comment by bhaugen** [comment]
I'll start another branch for https://github.com/FreedomCoop/valuenetwork/issues/136

**Comment by bhaugen** [comment]
I think I am done, or at least very close to done, with my work on this one-week sprint.

#136 will want a date range selector and pagination, but those features are probably not critical now.

So, what am I missing? What can I help other people with? What else should I do next?

Tomorrow Lynn and I are running homestead errands. Sunday I can do a bit more sprint work, but then I want to review the sprint:
* how did we do?
* how did you like working this way?
* should we keep working this way?
* if so, what's on for the next sprint?

**Comment by bhaugen** [comment]
@XaviP how should we finish things off for this sprint? Should we do pull requests for branches we think are finished? (In my case, I'm leaving branches ready to do _something_ but leaving style and nav choices to @bum2 in some cases. He may not want to finish them. If so, he can tell me which ones to do some finishing touches on.)

Or should we leave it to you to decide how to stage the branches through testocp and on to prod ocp?

**Comment by bhaugen** [comment]
One branch that could be trouble is request_actions. It has a migration that needs to be merged with the project branch, which might already have been merged into master. I can merge that one into master locally and work through the migration merge if you want.

**Comment by XaviP** [comment]
I think bumbum is working in (from a telegram conversation):

> I'm working about the join-a-project request process managed by project coordinators (managers), when project is 'moderated'... trying to integrate a flexible forms plugin... If the project chooses 'public' visibility the form shows in frontend without login. Similar of membership system but generic and hopefully flexible custom fields...

So, I think we can wait for this next feature, as it seems will be necessary for CVNs next week.
In the meanwhile, I think there are some simple issues marked as bug that can be worked if you want. Also, the suggested skills need some development to be aproved by admins.

These are some ideas, but now I cannot work on them for some days.

**Comment by bhaugen** [comment]
Here's what I think is on my plate now:
* https://github.com/FreedomCoop/valuenetwork/issues/64#issuecomment-244565947 take "other" fields through to admin creation of new skills
* https://github.com/FreedomCoop/valuenetwork/issues/137#issuecomment-244596739 merge the request_actions branch to master and deal with the migration merge conflict.
* https://github.com/FreedomCoop/valuenetwork/issues/107#issuecomment-244596788 button to create agent+user from the JoinRequest data automatically

What's the priority lineup? Anything else I should be doing?

**Comment by bhaugen** [comment]
Lynn and I discussed the priority question on our morning walk. We think the priority should be to consolidate all the work we have done so far: get everything that is ok for testocp merged into master, deal with all the migration conflicts, etc. Then put the remaining issues into the next sprint.

We're eating breakfast now, and I'll come back and merge the request_actions branch afterwards (unless those strong objections).

After that I got a one-hour-time-boxed task to help Lynn with some value flows work, then back here to help with any other consolidation and migration conflicts etc.

**Comment by bhaugen** [comment]
> After that I got a one-hour-time-boxed task to help Lynn with some value flows work, then back here to help with any other consolidation and migration conflicts etc.

Done with that task. I won't do anything else until @bum2 merges whatever he intends to merge into master at this stage of consolidation of our first sprint. Then I think we should summarize the sprint as I suggested above in [this comment](https://github.com/FreedomCoop/valuenetwork/issues/131#issuecomment-244479750). and then, assuming we all want to continue with this pattern, define the next sprint, which can start out with the missing pieces of this sprint.

**Comment by XaviP** [comment]
Deployed in testocp last merge to master:
https://github.com/FreedomCoop/valuenetwork/issues/137#issuecomment-244722330

**Comment by bhaugen** [comment]
> Here's what I think is on my plate now (copied from upstream comment):
> * https://github.com/FreedomCoop/valuenetwork/issues/64#issuecomment-244565947 take "other" fields through to admin creation of new skills
> * ~~https://github.com/FreedomCoop/valuenetwork/issues/137#issuecomment-244596739 merge the request_actions branch to master and deal with the migration merge conflict.~~ done
> * https://github.com/FreedomCoop/valuenetwork/issues/107#issuecomment-244596788 button to create agent+user from the JoinRequest data automatically

I don't want to take on the last one until @bum2 has merged the stuff he has been working on. I'll write up a rough design on the first one soon, but don't want to do that one until then, either, because that one will need yet another migration.

**Comment by bum2** [comment]
You are right Bob! Here i have updated the fobi branch, without that button, but all the rest is prepared (when the new agent is linked to the join_request). So only pending this button.

**Comment by bhaugen** [comment]
@bum2 where should I work on that? In the fobi branch? Or are you merging the fobi branch into master first?
Also, has the project branch been merged into master yet? I mean, is all of the sprint work in master now? Or is there anything waiting to be merged?

**Comment by bum2** [comment]
Project branch was merged long ago, master is merged into fobi and also the Lynn branch is merged too. Everything in master now. Is there any other thing to merge?

**Comment by XaviP** [comment]
This sprint is released in v0.9.4 https://github.com/FreedomCoop/valuenetwork/releases/tag/v0.9.4
Closing.
